pkg/api/v1/types.go

```diff
@@ -582,6 +582,8 @@ type RBDVolumeSource struct {
 	// Defaults to false.
 	// More info: http://releases.k8s.io/HEAD/examples/rbd/README.md#how-to-use-it
 	ReadOnly bool `json:"readOnly,omitempty"`
+	// Optional: Sidecar is the sidecar container's name
+	Sidecar string `json:"sidecar,omitempty"`
```

Comment:
Does the image name need to vary by volume or would a single image name
suffice for all instances of the type?

The new VolumeConfig stuff is meant to easily do the latter. An admin can
configure the image to use.

On Monday, September 21, 2015, Huamin Chen <notifications@github.com> wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/13691#discussion_r39969486>
> :
>
> > @@ -582,6 +582,8 @@ type RBDVolumeSource struct {
> >  	// Defaults to false.
> >  	// More info: http://releases.k8s.io/HEAD/examples/rbd/README.md#how-to-use-it
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> > +	// Optional: Sidecar is the sidecar container's name
> > +	Sidecar string `json:"sidecar,omitempty"`
>
> Sidecar is the image name. It is part of the pod API that user can specify
> so the appropriate container image can be started to execute the command.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13691/files#r39969486>.
>